module/gui/LinkDock.py

```diff
@@ -19,38 +19,210 @@
+                return
+            self.widget.box.setEnabled(False)
+            self.widget.filter.setEnabled(False)
+            self.undoText = text
+            self.emit(SIGNAL("parseUri"), "linkdock", unicode(text))
+        else:
+            self.widget.box.setPlainText(self.undoText)
+            self.undo = False
+            self.setFilterBtnText()
+    
+    def slotBoxTextChanged(self):
+        if self.undo and self.textChangeCount > 0:
+            self.undo = False
+            self.setFilterBtnText()
+        else:
+            self.textChangeCount += 1
+    
+    def parseUriResult(self, result):
+        self.textChangeCount = 0
+        self.widget.box.setPlainText(result)
+        self.undo = True
+        self.setFilterBtnText()
+        self.widget.box.setEnabled(True)
+        self.widget.filter.setEnabled(True)
+    
+    def slotClearBtnClicked(self):
         self.widget.box.clear()
+    
+    def defaultSettings(self):
+        self.widget.append.setChecked(True)
+        self.widget.destQueue.setChecked(False)
+        self.widget.destCollector.setChecked(True)
+        self.widget.destAutoSelect.setChecked(True)
+    
+    def getSettings(self):
+        settings = {}
+        #settings["append"]     = self.widget.append.isChecked()
+        settings["queue"]      = self.widget.destQueue.isChecked()
+        settings["autoSelect"] = self.widget.destAutoSelect.isChecked()
+        return settings
+    
+    def setSettings(self, settings):
+        #self.widget.append.setChecked         (settings["append"])
+        self.widget.destQueue.setChecked      (settings["queue"])
+        self.widget.destCollector.setChecked  (not settings["queue"])
+        self.widget.destAutoSelect.setChecked (settings["autoSelect"])
+    
+    def closeEvent(self, event):
+        if self.isFloating():
+            self.emit(SIGNAL("newLinkDockClosed"))
         self.hide()
+        event.ignore()
+    
+    def paintEvent(self, event):
+        QDockWidget.paintEvent(self, event)
+        if self.isFloating():
+            self.geo = self.geometry()
+        self.paintEventCounter += 1
+        if self.paintEventSignal:
+            self.paintEventSignal = False
+            self.emit(SIGNAL("newLinkDockPaintEvent"))
+    
+    def moveEvent(self, event):
+        if self.isFloating():
+            self.geo = self.geometry()
 
 class NewLinkWindow(QWidget):
     def __init__(self, dock):
         QWidget.__init__(self)
+        self.log = logging.getLogger("guilog")
         self.dock = dock
+        
         self.setLayout(QVBoxLayout())
         layout = self.layout()
         
-        explanationLabel = QLabel("Select a package and then click Add button.")
-        boxLabel = QLabel("Paste URLs here:")
-        self.box = QTextEdit()
+        boxLabel = QLabel(_("Drop or Paste URLs here") + ":")
+        self.box = PlainTextEdit()
         
-        save = QPushButton("Add")
+        self.append = QCheckBox(_("Append URLs"))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['QCheckBox' may be undefined, or defined from star imports: PyQt4.QtCore, PyQt4.QtGui](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)